make/modules/jdk.jpackage/Lib.gmk

```diff
@@ -44,27 +58,59 @@ endif
 
 JPACKAGE_OUTPUT_DIR := $(JDK_OUTPUTDIR)/modules/$(MODULE)/jdk/jpackage/internal/resources
 JPACKAGE_CXXFLAGS_windows := -EHsc -DUNICODE -D_UNICODE
+JPACKAGE_CFLAGS_windows := -DUNICODE -D_UNICODE
+JPACKAGE_APPLAUNCHER_INCLUDES := $(addprefix -I, \
+    $(sort $(JPACKAGE_APPLAUNCHER_SRC) $(JPACKAGE_APPLAUNCHERLIB_SRC)) \
+)
 
 # Output app launcher executable in resources dir, and symbols in the object dir
 $(eval $(call SetupJdkExecutable, BUILD_JPACKAGE_APPLAUNCHEREXE, \
     NAME := jpackageapplauncher, \
     OUTPUT_DIR := $(JPACKAGE_OUTPUT_DIR), \
     SYMBOLS_DIR := $(SUPPORT_OUTPUTDIR)/native/$(MODULE)/jpackageapplauncher, \
     SRC := $(JPACKAGE_APPLAUNCHER_SRC), \
-    TOOLCHAIN := TOOLCHAIN_LINK_CXX, \
+    TOOLCHAIN := $(JPACKAGE_APPLAUNCHER_TOOLCHAIN), \
     OPTIMIZATION := LOW, \
     CXXFLAGS := $(call JpackageWithStaticCrt, $(CXXFLAGS_JDKEXE)) \
-        $(addprefix -I, $(JPACKAGE_APPLAUNCHER_SRC)), \
+        $(JPACKAGE_APPLAUNCHER_INCLUDES), \
+    CFLAGS := $(call JpackageWithStaticCrt, $(CFLAGS_JDKEXE)) \
+        $(JPACKAGE_APPLAUNCHER_INCLUDES), \
+    CFLAGS_linux := -Wno-format-nonliteral, \
+    CFLAGS_macosx := -Wno-format-nonliteral, \
     CXXFLAGS_windows := $(JPACKAGE_CXXFLAGS_windows), \
+    CFLAGS_windows := $(JPACKAGE_CFLAGS_windows), \
     LDFLAGS := $(LDFLAGS_JDKEXE), \
-    LIBS_macosx := -framework Cocoa, \
-    LIBS := $(LIBCXX), \
-    LIBS_linux := -ldl, \
+    LIBS_macosx := $(LIBCXX) -framework Cocoa, \
+    LIBS_windows := $(LIBCXX), \
+    LIBS_linux := -nodefaultlibs -lc -ldl, \
 ))
 
-$(BUILD_JPACKAGE_APPLAUNCHEREXE): $(call FindLib, java.base, java)
```

Comment:
Why did you remove this dependency?

Reply:
I moved it to the bottom of the file making all artifacts produced by make/modules/jdk.jpackage/Lib.gmk depend on java.bas and java. There is `$(JPACKAGE_TARGETS): $(call FindLib, java.base, java)` at the bottom of the file.